### PR TITLE
Stage IV: Add initial report

### PR DESCRIPTION
**Pull request by JacekGlen**

### CODE REVIEW

File: docs/report_stage_iv.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+### Measured OPCODEs and precompiles
+In this stage, we finally measure all* OPCODEs together with the precompiles.
+
+### Tooling and automation *
+The reproducibility is the key to the research. We have provided a complete setup guide and tooling to make the execution of the benchmarks as easy as possible. There are two ways to perform the measurements in your environment. The first one is to use the provided scripts to build the EVM implementations and run the benchmarks. The second one is to use the provided release with precompiled binaries. The binaries are available for Linux x64, MacOS x64 and Windows.
+
+## Methodology
+
+### Measurement approach
+Our approach is to test each EVM implementation in isolation. That means that any host objects, storage access and other infrastructure elements are either mocked, or a minimal implementation is used. This allows us to measure the execution time of the OPCODEs directly, without any additional overhead.
```
Comment (lukasz-glen) [inline docs/report_stage_iv.md:32]: Something is not right . I have a feeling that each OPCODE is measured as one execution, independently. Maybe this additional sentence will clarify:
Measuring a single OPCODE execution time is inaccurate and susceptible to external factors. For that reason larger bytecode programs are executed and OPCODEs costs are estimated using statistic tools.

File: docs/report_stage_iv.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
+# Gas Cost Estimator
+_Stage 4 Report_
+
+> **_NOTE:_**  This document is a work in progress. The * denotes incomplete parts.
+
+## Abstract
+In this stage, we use the findings from the previous stages and apply them to produce a comprehensive analysis of the gas cost. The improved methodology incorporates standardized benchmarks, data analysis and report generation. The scope has been extended to all OPCODEs, precompiles and top 7 EVM implementations. The reproducibility of the results has been improved by providing a complete setup guide and tooling. Additionally, the release contains precompiled binaries for even easier execution. The result of this stage is a proposal for a new Gas Cost Scheduled to be included in the next hard fork.
```
[the remark this answers is not in the document]
Comment (JacekGlen) [reply docs/report_stage_iv.md]: Done

[the remark this answers is not in the document]
Comment (JacekGlen) [reply docs/report_stage_iv.md]: Done

File: docs/report_stage_iv.md (hunk 1 above)
Comment (JacekGlen) [reply docs/report_stage_iv.md]: Added explanation

**Pull request merged**